controllers/v1/users.js

```diff
@@ -591,5 +591,177 @@ module.exports = class Users extends Abstract {
+  * @apiVersion 1.0.0
+  * @apiName User solutions
+  * @apiGroup Users
+  * @apiHeader {String} internal-access-token Internal access token
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
+  * @apiSampleRequest /kendra/api/v1/users/solutions/5ff438b04698083dbfab7284?page=1&limit=10
+  * @apiParamExample {json} Request-Body:
+  * {
+        "role" : "HM",
+   		"state" : "236f5cff-c9af-4366-b0b6-253a1789766a",
+        "district" : "1dcbc362-ec4c-4559-9081-e0c2864c2931",
+        "school" : "c5726207-4f9f-4f45-91f1-3e9e8e84d824"
+    }
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
+  * {
+    "message": "Successfully fetched user targeted solution",
+    "status": 200,
+    "result": {
+        "data": [
+            {
+                "_id": "5fc3dff14ea9b44f3340afe2",
+                "programId": "5ff438b04698083dbfab7284",
+                "programName": "TEST_SCOPE_PROGRAM",
+                "externalId": "f449823a-06bb-4a3f-9d49-edbe1524ebbb-1606672337956"
+            },
+            {
+                "_id": "5ff447e127ef425953bd8306",
+                "externalId": "TEST_SOLUTION_SCOPE",
+                "programId": "5ff438b04698083dbfab7284",
+                "programName": "TEST scope in program"
+            },
+            {
+                "_id": "5ff482737f768d2de902e912",
+                "externalId": "SCOPE_OBSERVATION_TEST",
+                "name": "observation testing",
+                "description": "Testing observation",
+                "programId": "5ff438b04698083dbfab7284",
+                "programName": "TEST scope in program"
+            }
+        ],
+        "count": 3,
+        "programName": "TEST_SCOPE_PROGRAM",
+        "description": "View and participate in educational programs active in your location and designed for your role."
+    }
+    }
+  **/
+
+  /**
+  * User targeted solutions.
+  * @method
+  * @name solutions
+  * @param  {req}  - requested data.
+  * @returns {json} List of targeted solutions.
+  */
+
+ solutions(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let targetedSolutions = await usersHelper.solutions(
+            req.params._id,
+            req.body,
+            req.userDetails.userToken,
+            req.pageSize,
+            req.pageNo,
+            req.searchText
+        );
+
+        targetedSolutions["result"] = targetedSolutions.data;
+        return resolve(targetedSolutions);
+
+      } catch (error) {
+
+        return reject({
+            status: 
+            error.status || 
+            httpStatusCode["internal_server_error"].status,
+
+            message: 
+            error.message || 
+            httpStatusCode["internal_server_error"].message
+        })
+
+      }
+    });
+  }
+
+
+/**
+     * @api {post} /kendra/api/v1/users/programs?page=:page&limit=:limit&search=:search 
```

Comment:
@aman-tunerlabs - We can have userId being passed in query parameter which can be given priority over token userId and only if the internal access token is given, will help in debugging in future.